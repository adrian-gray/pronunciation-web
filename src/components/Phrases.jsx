import React from "react";

import Paper from "@material-ui/core/Paper";
import Table from "@material-ui/core/Table";
import TableBody from "@material-ui/core/TableBody";
import TableCell from "@material-ui/core/TableCell";
import TableRow from "@material-ui/core/TableRow";
import Typography from "@material-ui/core/Typography";
import { makeStyles } from "@material-ui/core/styles";
import { withTheme } from "@material-ui/styles";

import SplitHilite from "./SplitHilite";

const useStyles = makeStyles(theme => ({
  headspace: theme.headspace,
  subCanvas: theme.subCanvas,
  largeText: theme.largeText
}));

const Phrases = props => {
  const classes = useStyles(props);

  const { ipa, phrases } = props;

  const list = Object.entries(phrases).map((arr, index) => {
    return (
      <TableRow key={index}>
        <TableCell>
          <Typography className={classes.largeText}>
            <SplitHilite str={arr[0]} />
          </Typography>
        </TableCell>
        <TableCell>
          <Typography className={classes.largeText}>
            <SplitHilite str={arr[1]} />
          </Typography>
        </TableCell>
      </TableRow>
    );
  });

  return (
    <div className={classes.headspace}>
      <Typography variant="h5" gutterBottom>
        {`Phrases using `}
        <SplitHilite str={ipa} />
      </Typography>

      <Paper className={classes.subCanvas}>
        <Table>
          <TableBody>{list}</TableBody>
        </Table>
      </Paper>
    </div>
  );
};

export default withTheme(Phrases);
