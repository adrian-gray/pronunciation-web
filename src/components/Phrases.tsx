import React from "react";

import Container from "react-bootstrap/Container";
import Table from "react-bootstrap/Table";

import SplitHilite from "./SplitHilite";

export default props => {
  const { ipa, phrases } = props;

  const list = Object.entries(phrases).map((arr, index) => {
    return (
      <tr key={index}>
        <td>
          <p className="large-text">
            <SplitHilite str={arr[0]} />
          </p>
        </td>
        <td>
          <p className="large-text">
            <SplitHilite str={arr[1]} />
          </p>
        </td>
      </tr>
    );
  });

  return (
    <div className="head-space">
      <h3>
        {`Phrases using `}
        <SplitHilite str={ipa} />
      </h3>

      <Container className="sub-canvas">
        <Table>
          <tbody>{list}</tbody>
        </Table>
      </Container>
    </div>
  );
};
