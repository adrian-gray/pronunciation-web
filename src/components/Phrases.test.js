import React from "react";
import { shallow } from "enzyme";
import { render, cleanup } from "@testing-library/react";

import Phrases from "./Phrases";

afterEach(cleanup);

test("Phrases", async () => {
  const props = {};

  const container = shallow(<Phrases {...props} />);

  expect(container.firstChild).toMatchSnapshot();
});
