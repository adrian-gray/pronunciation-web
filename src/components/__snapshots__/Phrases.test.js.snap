// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Phrases 1`] = `undefined`;
